src/java/org/apache/cassandra/db/memtable/SkipListMemtable.java

```diff
@@ -273,7 +273,7 @@ private static int estimateRowOverhead(final int count)
             final Object val = new Object();
             for (int i = 0 ; i < count ; i++)
                 partitions.put(cloner.clone(new BufferDecoratedKey(new LongToken(i), HeapCloner.instance.allocate(1))), val);
-            double avgSize = ObjectSizes.measureDeep(partitions) / (double) count;
+            double avgSize = ObjectSizes.measureDeepOmitShared(partitions) / (double) count;
```

Comment:
This is why `MemtableSizeHeapBuffersTest` was failing.

Until now, `ObjectSizes.meter` was built with `omitSharedBufferOverhead()`. But with the changes in this PR, it isn't, so we need a dedicated method (copied from [49e0c611](https://github.com/datastax/cassandra/commit/49e0c61107005b1a83799f7f1e6c0a855d159c29#diff-42c5b9b54e9a13dfe0cb013d45725c3cc4011648e148d88a34173afa1816d74f) on the main-5.0 branch).

FWIW, adding logs in `SkipListmemtable#estimateRowOverhead()` helped pinpoint the issue, the estimated sizes between main and this PR were wildly different.

Reply:
> FWIW, adding logs in SkipListmemtable#estimateRowOverhead() helped pinpoint the issue, the estimated sizes between main and this PR were wildly different.

For debugging purposes, you can also use `.printVisitedTree()` to get a printout of all the objects
and their calculated size on heap to debug jamm related issues. See:
<a href="https://github.com/jbellis/jamm/blob/master/README.md#visited-object-tree">README.md on github</a>